### DISCUSSION TITLE
Add support for multilevel hierarchies containing columns from a partially joined table

### DISCUSSION THREAD

**Discussion by APolyakova** [atoti/atoti#627] [General]
I would like to create a multi-level hierarchy with levels: FromCurveId and AlternativeCurveId. However I'm getting an exception:  Cannot create hierarchy Curve Relations because the fields [FromCurveId, AlternativeCurveId] cannot be found

@fabiencelier referred me to the documentation: From the join documentation:
Only some of the key columns of the destination store are mapped: this is a partial reference:
The columns from the source store used in the mapping must be attached to hierarchies.

I have read the last sentence 5 times, but I still don't understand why I can create a multi-level hierarchy. I can put the fields onto the rows, and they work fine. Please advise! Thank you

[can't create a hierarchy.ipynb.zip](https://github.com/atoti/atoti/files/5532633/can.t.create.a.hierarchy.ipynb.zip)
![image (2)](https://user-images.githubusercontent.com/37184538/98983292-6a995600-24d5-11eb-9f2c-c1cfb2861de4.png)

**Comment by tibdex** [comment]
Hi Anastasia, as seen on the schema, the join between `Sensitivities` and  `Alternative Curves` is partial and multi hierarchies cannot contain a level coming from a column of a partially joined store.

This should be supported in the future.

**Comment by isobelbyrne** [comment]
Hi team, has support for this been added yet? I seem to be having the same issue currently in 0.7.2, where I need to create a multi-level hierarchy from from an account info table partially joined to the base store, however I receive the following error:
`AtotiJavaException: Cannot create hierarchy Account Hierarchy because the fields [StoreField(store=account_Table, name=Level 1), StoreField(store=account_Table, name=Level 2), StoreField(store=account_Table, name=Level 3), StoreField(store=account_Table, name=Level 4), StoreField(store=account_Table, name=Level 5), StoreField(store=account_Table, name=Level 6)] cannot be found.`

Thanks for your help!

**Reply by tibdex** [reply]
Hi, multilevel hierarchies on partially joined columns are still not supported. Though we do want to support them, it's not in the short term roadmap.

**Reply by isobelbyrne** [reply]
Thanks for the update! The issue I am facing here is that I would like to include some marginal contribution calculations along the chart of accounts hierarchy for a PoC currently on-going. I will have a think how I can achieve this now without the accounts hierarchy, but do let me know when a supporting release can be expected.

**Comment by mosghaier** [comment]
Hey team,

One of our prospects is running an internal PoC testing out Atoti for different use cases and they have reported running into this very issue. Are we any closer to getting support for it added? Do we have a timeline? 

Thank you!

**Reply by tibdex** [reply]
Hi, no progress has been made on this feature. We can't provide a timeline at the moment.

**Reply by mosghaier** [reply]
Alright, thank you for the intel!

**Reply by rcolle** [reply]
@mosghaier can we discuss this internally through JIRA or slack? Is that a blocker or can we work around the issue?

**Reply by tibdex** [reply]
You could comment on [this internal ticket](https://activeviam.atlassian.net/browse/PYTHON-19) for instance.
